create-convex/src/packageManagers.ts

```diff
@@ -0,0 +1,47 @@
+import spawn from "cross-spawn";
+
+// Type definition for package managers (no unknown - defaults to npm)
+export type PackageManager = "npm" | "pnpm" | "yarn" | "bun";
+
+/**
+ * Detects which package manager is currently being used
+ * First checks environment variables, then process.versions for Bun,
+ * and falls back to checking which package managers are installed
+ * @returns The detected package manager, defaulting to "npm" if none detected
+ */
+export function detectPackageManager(): PackageManager {
+  // First check npm_config_user_agent environment variable
+  const userAgent = process.env.npm_config_user_agent;
+  if (userAgent) {
+    if (userAgent.includes("yarn")) return "yarn";
+    if (userAgent.includes("pnpm")) return "pnpm";
+    if (userAgent.includes("bun")) return "bun";
+    if (userAgent.includes("npm")) return "npm";
+  }
+
+  // Check if Bun is being used via process.versions
+  if (process.versions.bun) return "bun";
+
+  // Fallback to checking for commands in PATH
+  try {
+    // Try to run each package manager with --version flag to see if it's installed
+    const checkCommand = (cmd: string): boolean => {
+      try {
+        const result = spawn.sync(cmd, ["--version"], { stdio: "ignore" });
+        return result.status === 0;
+      } catch (error) {
+        return false;
+      }
+    };
+
+    if (checkCommand("bun")) return "bun";

```

Comment:
Does this no `npm_config_user_agent` env var set condition happen? Since it's already an unexpected case how about returning npm here instead, if it does happen I wouldn't want globably installing bun to break the a workflow

Reply:
haha sorry that useAgent code is disgusting, ill get that tidied up.

Okay so dont do that "checkCommand" part? Instead just default to NPM in that case?

Reply:
Sure either don't do the checkCommand or do it, but check for npm first